lib/assembly/server.ex

```diff
@@ -1,24 +1,33 @@
 defmodule Assembly.Server do
   use GRPC.Server, service: Bottle.Assembly.V1.Service
 
-  alias Assembly.{Repo, Schemas}
-  alias Bottle.Assembly.V1.{Build, BuildListRequest, BuildListResponse}
+  import Ecto.Query
 
-  @spec build_list(BuildListRequest.t(), GRPC.Server.Stream.t()) :: BuildListResponse.t()
-  def build_list(_request, _stream) do
-    builds =
-      Schema.Build
-      |> Repo.all()
-      |> Enum.map(&bottled_build/1)
+  alias Assembly.{Caster, Repo, Schemas}
+  alias Bottle.Assembly.V1.{BuildListRequest, BuildListResponse}
 
-    BuildListResponse.new(builds: builds)
+  @spec build_list(BuildListRequest.t(), GRPC.Server.Stream.t()) :: any
+  def build_list(_request, stream) do
+    query =
+      from b in Schemas.Build,
+        left_join: c in assoc(b, :build_components),
+        where: b.status != :built,
+        preload: [build_components: c]
```

Comment:
Right now this is based on DB lookup. I believe we want to shift to the supervisor here instead. It's less hits to the DB and it's more updated info than the DB as the persistence may not have occurred yet.